File: .conver/ui/screens/SellerScreen.kt
package com.example.ui.screens

import androidx.compose.animation.*
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.data.model.Seller
import com.example.ui.theme.TealTertiary
import com.example.ui.viewmodel.AppViewModel

@Composable
fun SellerScreen(
    viewModel: AppViewModel,
    modifier: Modifier = Modifier
) {
    val sellers by viewModel.sellers.collectAsState()
    val searchQuery by viewModel.sellerSearchQuery.collectAsState()
    val filterStatus by viewModel.sellerFilterStatus.collectAsState()

    var showAddSellerDialog by remember { mutableStateOf(false) }
    var sellerToBan by remember { mutableStateOf<Seller?>(null) }

    // Filter and search computation
    val filteredSellers = remember(sellers, searchQuery, filterStatus) {
        sellers.filter { seller ->
            val matchQuery = seller.name.contains(searchQuery, ignoreCase = true) ||
                    seller.storeName.contains(searchQuery, ignoreCase = true) ||
                    seller.email.contains(searchQuery, ignoreCase = true)

            val matchStatus = when (filterStatus) {
                "Aktif" -> seller.status == "Aktif" && !seller.isBanned
                "Tidak Aktif" -> seller.status == "Tidak Aktif" && !seller.isBanned
                "Banned" -> seller.isBanned
                else -> true // "Semua"
            }

            matchQuery && matchStatus
        }
    }

    Column(
        modifier = modifier
            .fillMaxSize()
            .padding(horizontal = 16.dp)
    ) {
        // Search and Add Header Row
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(vertical = 12.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            OutlinedTextField(
                value = searchQuery,
                onValueChange = { viewModel.setSellerSearchQuery(it) },
                placeholder = { Text("Cari nama, toko, atau email...") },
                leadingIcon = { Icon(Icons.Default.Search, contentDescription = "Search Icon") },
                singleLine = true,
                shape = RoundedCornerShape(12.dp),
                colors = OutlinedTextFieldDefaults.colors(
                    focusedBorderColor = MaterialTheme.colorScheme.primary,
                    unfocusedBorderColor = MaterialTheme.colorScheme.surfaceVariant
                ),
                modifier = Modifier
                    .weight(1f)
                    .testTag("seller_search_input")
            )

            FloatingActionButton(
                onClick = { showAddSellerDialog = true },
                containerColor = MaterialTheme.colorScheme.primary,
                contentColor = MaterialTheme.colorScheme.onPrimary,
                shape = RoundedCornerShape(12.dp),
                modifier = Modifier
                    .size(56.dp)
                    .testTag("add_seller_fab")
            ) {
                Icon(Icons.Default.PersonAdd, contentDescription = "Daftarkan Seller")
            }
        }

        // Filter chips list
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(bottom = 8.dp),
            horizontalArrangement = Arrangement.spacedBy(8.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            val filters = listOf("Semua", "Aktif", "Tidak Aktif", "Banned")
            filters.forEach { status ->
                val selected = filterStatus == status
                FilterChip(
                    selected = selected,
                    onClick = { viewModel.setSellerFilterStatus(status) },
                    label = { Text(status) },
                    colors = FilterChipDefaults.filterChipColors(
                        selectedContainerColor = MaterialTheme.colorScheme.primary.copy(alpha = 0.15f),
                        selectedLabelColor = MaterialTheme.colorScheme.primary,
                        selectedLeadingIconColor = MaterialTheme.colorScheme.primary
                    ),
                    modifier = Modifier.testTag("filter_chip_$status")
                )
            }
        }

        // Sellers count indicator
        Text(
            text = "Ditemukan ${filteredSellers.size} Penjual",
            style = MaterialTheme.typography.bodySmall,
            fontWeight = FontWeight.Bold,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
            modifier = Modifier.padding(bottom = 12.dp)
        )

        // Sellers List
        if (filteredSellers.isEmpty()) {
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .weight(1f),
                contentAlignment = Alignment.Center
            ) {
                Column(horizontalAlignment = Alignment.CenterHorizontally) {
                    Icon(
                        Icons.Default.PeopleOutline,
                        contentDescription = "Empty",
                        tint = MaterialTheme.colorScheme.onSurfaceVariant,
                        modifier = Modifier.size(64.dp)
                    )
                    Spacer(modifier = Modifier.height(8.dp))
                    Text(
                        text = "Tidak ada penjual yang cocok",
                        color = MaterialTheme.colorScheme.onSurfaceVariant,
                        style = MaterialTheme.typography.bodyMedium
                    )
                }
            }
        } else {
            LazyColumn(
                modifier = Modifier.weight(1f),
                verticalArrangement = Arrangement.spacedBy(12.dp),
                contentPadding = PaddingValues(bottom = 24.dp)
            ) {
                items(filteredSellers, key = { it.id }) { seller ->
                    SellerRowItem(
                        seller = seller,
                        onBanClick = { sellerToBan = seller },
                        onUnbanClick = { viewModel.unbanSeller(seller.id) },
                        onDeleteClick = { viewModel.deleteSeller(seller.id) }
                    )
                }
            }
        }
    }

    if (showAddSellerDialog) {
        AddSellerDialog(
            onDismiss = { showAddSellerDialog = false },
            onConfirm = { name, store, email, phone ->
                viewModel.addNewSeller(name, store, email, phone)
                showAddSellerDialog = false
            }
        )
    }

    if (sellerToBan != null) {
        BanReasonDialog(
            seller = sellerToBan!!,
            onDismiss = { sellerToBan = null },
            onConfirm = { reason ->
                viewModel.banSeller(sellerToBan!!.id, reason)
                sellerToBan = null
            }
        )
    }
}

@Composable
fun SellerRowItem(
    seller: Seller,
    onBanClick: () -> Unit,
    onUnbanClick: () -> Unit,
    onDeleteClick: () -> Unit
) {
    var expandedMenu by remember { mutableStateOf(false) }

    Card(
        modifier = Modifier
            .fillMaxWidth()
            .shadow(1.dp, RoundedCornerShape(16.dp)),
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(
            containerColor = if (seller.isBanned) {
                MaterialTheme.colorScheme.errorContainer.copy(alpha = 0.1f)
            } else {
                MaterialTheme.colorScheme.surface
            }
        ),
        border = BorderStroke(
            width = 1.dp,
            color = if (seller.isBanned) {
                MaterialTheme.colorScheme.error.copy(alpha = 0.3f)
            } else {
                MaterialTheme.colorScheme.surfaceVariant
            }
        )
    ) {
        Column(modifier = Modifier.padding(16.dp)) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                verticalAlignment = Alignment.Top,
                horizontalArrangement = Arrangement.SpaceBetween
            ) {
                Row(
                    verticalAlignment = Alignment.CenterVertically,
                    modifier = Modifier.weight(1f)
                ) {
                    // Profile Icon placeholder with dynamic alphabet
                    Box(
                        modifier = Modifier
                            .size(48.dp)
                            .background(
                                color = if (seller.isBanned) {
                                    MaterialTheme.colorScheme.error.copy(alpha = 0.2f)
                                } else if (seller.status == "Aktif") {
                                    TealTertiary.copy(alpha = 0.15f)
                                } else {
                                    MaterialTheme.colorScheme.surfaceVariant
                                },
                                shape = CircleShape
                            ),
                        contentAlignment = Alignment.Center
                    ) {
                        Text(
                            text = seller.name.take(1).uppercase(),
                            style = MaterialTheme.typography.titleMedium,
                            fontWeight = FontWeight.Bold,
                            color = if (seller.isBanned) {
                                MaterialTheme.colorScheme.error
                            } else if (seller.status == "Aktif") {
                                TealTertiary
                            } else {
                                MaterialTheme.colorScheme.onSurfaceVariant
                            }
                        )
                    }
                    Spacer(modifier = Modifier.width(16.dp))
                    Column {
                        Text(
                            text = seller.name,
                            style = MaterialTheme.typography.titleSmall,
                            fontWeight = FontWeight.Bold,
                            color = MaterialTheme.colorScheme.onSurface
                        )
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Icon(
                                Icons.Default.Storefront,
                                contentDescription = "Toko",
                                tint = MaterialTheme.colorScheme.primary,
                                modifier = Modifier.size(14.dp)
                            )
                            Spacer(modifier = Modifier.width(4.dp))
                            Text(
                                text = seller.storeName,
                                style = MaterialTheme.typography.bodySmall,
                                fontWeight = FontWeight.Bold,
                                color = MaterialTheme.colorScheme.onSurfaceVariant
                            )
                        }
                    }
                }

                // Action Menu Dropdown
                Box {
                    IconButton(onClick = { expandedMenu = !expandedMenu }) {
                        Icon(Icons.Default.MoreVert, contentDescription = "Menu Aksi")
                    }
                    DropdownMenu(
                        expanded = expandedMenu,
                        onDismissRequest = { expandedMenu = false }
                    ) {
                        if (seller.isBanned) {
                            DropdownMenuItem(
                                text = { Text("Unban Seller") },
                                leadingIcon = { Icon(Icons.Default.CheckCircle, contentDescription = "Unban", tint = TealTertiary) },
                                onClick = {
                                    onUnbanClick()
                                    expandedMenu = false
                                }
                            )
                        } else {
                            DropdownMenuItem(
                                text = { Text("Ban Seller", color = MaterialTheme.colorScheme.error) },
                                leadingIcon = { Icon(Icons.Default.Block, contentDescription = "Ban", tint = MaterialTheme.colorScheme.error) },
                                onClick = {
                                    onBanClick()
                                    expandedMenu = false
                                }
                            )
                        }
                        DropdownMenuItem(
                            text = { Text("Hapus Permanen", color = MaterialTheme.colorScheme.error) },
                            leadingIcon = { Icon(Icons.Default.Delete, contentDescription = "Hapus", tint = MaterialTheme.colorScheme.error) },
                            onClick = {
                                onDeleteClick()
                                expandedMenu = false
                            }
                        )
                    }
                }
            }

            Spacer(modifier = Modifier.height(12.dp))

            // Extra Info Grid
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Column(verticalArrangement = Arrangement.spacedBy(4.dp)) {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(Icons.Default.Email, contentDescription = "Email", tint = MaterialTheme.colorScheme.onSurfaceVariant, modifier = Modifier.size(12.dp))
                        Spacer(modifier = Modifier.width(6.dp))
                        Text(seller.email, style = MaterialTheme.typography.bodySmall, color = MaterialTheme.colorScheme.onSurfaceVariant)
                    }
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(Icons.Default.Phone, contentDescription = "Telepon", tint = MaterialTheme.colorScheme.onSurfaceVariant, modifier = Modifier.size(12.dp))
                        Spacer(modifier = Modifier.width(6.dp))
                        Text(seller.contact, style = MaterialTheme.typography.bodySmall, color = MaterialTheme.colorScheme.onSurfaceVariant)
                    }
                }

                // Status Tag Badge
                Box(
                    modifier = Modifier
                        .clip(RoundedCornerShape(8.dp))
                        .background(
                            color = if (seller.isBanned) {
                                MaterialTheme.colorScheme.error.copy(alpha = 0.2f)
                            } else if (seller.status == "Aktif") {
                                TealTertiary.copy(alpha = 0.2f)
                            } else {
                                MaterialTheme.colorScheme.surfaceVariant
                            }
                        )
                        .padding(horizontal = 10.dp, vertical = 6.dp)
                ) {
                    Text(
                        text = if (seller.isBanned) "BANNED" else seller.status.uppercase(),
                        style = MaterialTheme.typography.labelSmall,
                        fontWeight = FontWeight.Bold,
                        color = if (seller.isBanned) {
                            MaterialTheme.colorScheme.error
                        } else if (seller.status == "Aktif") {
                            TealTertiary
                        } else {
                            MaterialTheme.colorScheme.onSurfaceVariant
                        }
                    )
                }
            }

            // Reason for banned (if exists)
            if (seller.isBanned && seller.banReason != null) {
                Spacer(modifier = Modifier.height(8.dp))
                Box(
                    modifier = Modifier
                        .fillMaxWidth()
                        .background(
                            MaterialTheme.colorScheme.error.copy(alpha = 0.05f),
                            RoundedCornerShape(8.dp)
                        )
                        .padding(10.dp)
                ) {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(
                            Icons.Default.Warning,
                            contentDescription = "Warning",
                            tint = MaterialTheme.colorScheme.error,
                            modifier = Modifier.size(14.dp)
                        )
                        Spacer(modifier = Modifier.width(6.dp))
                        Text(
                            text = "Alasan: ${seller.banReason}",
                            style = MaterialTheme.typography.bodySmall,
                            fontSize = 11.sp,
                            color = MaterialTheme.colorScheme.error,
                            maxLines = 2,
                            overflow = TextOverflow.Ellipsis
                        )
                    }
                }
            }
        }
    }
}

@Composable
fun AddSellerDialog(
    onDismiss: () -> Unit,
    onConfirm: (String, String, String, String) -> Unit
) {
    var name by remember { mutableStateOf("") }
    var storeName by remember { mutableStateOf("") }
    var email by remember { mutableStateOf("") }
    var phone by remember { mutableStateOf("") }

    AlertDialog(
        onDismissRequest = onDismiss,
        title = {
            Text("Daftarkan Seller Baru", fontWeight = FontWeight.Bold)
        },
        text = {
            Column(
                verticalArrangement = Arrangement.spacedBy(12.dp),
                modifier = Modifier.fillMaxWidth()
            ) {
                OutlinedTextField(
                    value = name,
                    onValueChange = { name = it },
                    label = { Text("Nama Lengkap") },
                    placeholder = { Text("Contoh: Rudi Hermawan") },
                    singleLine = true,
                    modifier = Modifier.fillMaxWidth().testTag("seller_name_input")
                )

                OutlinedTextField(
                    value = storeName,
                    onValueChange = { storeName = it },
                    label = { Text("Nama Toko") },
                    placeholder = { Text("Contoh: Jaya Cellular") },
                    singleLine = true,
                    modifier = Modifier.fillMaxWidth().testTag("seller_store_input")
                )

                OutlinedTextField(
                    value = email,
                    onValueChange = { email = it },
                    label = { Text("Email") },
                    placeholder = { Text("Contoh: rudi.hermawan@gmail.com") },
                    singleLine = true,
                    modifier = Modifier.fillMaxWidth().testTag("seller_email_input")
                )

                OutlinedTextField(
                    value = phone,
                    onValueChange = { phone = it },
                    label = { Text("Nomor Telepon / WhatsApp") },
                    placeholder = { Text("Contoh: 0812345678") },
                    singleLine = true,
                    modifier = Modifier.fillMaxWidth().testTag("seller_phone_input")
                )
            }
        },
        confirmButton = {
            Button(
                onClick = {
                    if (name.isNotBlank() && storeName.isNotBlank() && email.isNotBlank()) {
                        onConfirm(name, storeName, email, phone)
                    }
                },
                enabled = name.isNotBlank() && storeName.isNotBlank() && email.isNotBlank()
            ) {
                Text("Daftarkan")
            }
        },
        dismissButton = {
            TextButton(onClick = onDismiss) {
                Text("Batal")
            }
        }
    )
}

@Composable
fun BanReasonDialog(
    seller: Seller,
    onDismiss: () -> Unit,
    onConfirm: (String) -> Unit
) {
    var reason by remember { mutableStateOf("") }

    AlertDialog(
        onDismissRequest = onDismiss,
        title = {
            Text("Tangguhkan Seller", fontWeight = FontWeight.Bold)
        },
        text = {
            Column(modifier = Modifier.fillMaxWidth()) {
                Text(
                    text = "Konfirmasi pemblokiran toko '${seller.storeName}' oleh seller '${seller.name}'. Berikan alasan pemblokiran agar dicatat sistem.",
                    style = MaterialTheme.typography.bodyMedium,
                    modifier = Modifier.padding(bottom = 12.dp)
                )

                OutlinedTextField(
                    value = reason,
                    onValueChange = { reason = it },
                    label = { Text("Alasan Pelanggaran") },
                    placeholder = { Text("Contoh: Penipuan transaksi barang tiruan...") },
                    modifier = Modifier.fillMaxWidth().testTag("ban_reason_input")
                )
            }
        },
        confirmButton = {
            Button(
                onClick = {
                    if (reason.isNotBlank()) {
                        onConfirm(reason)
                    }
                },
                colors = ButtonDefaults.buttonColors(containerColor = MaterialTheme.colorScheme.error),
                enabled = reason.isNotBlank()
            ) {
                Text("Tangguhkan")
            }
        },
        dismissButton = {
            TextButton(onClick = onDismiss) {
                Text("Batal")
            }
        }
    )
}

File: lib/screens/seller_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../models/seller.dart';
import '../viewmodels/app_view_model.dart';

class SellerScreen extends StatelessWidget {
  const SellerScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Theme.of(context).scaffoldBackgroundColor,
      body: Padding(
        padding: const EdgeInsets.symmetric(horizontal: 16.0),
        child: Column(
          children: [
            const SizedBox(height: kToolbarHeight - 20), // Top padding
            _SearchAndAddHeader(),
            const SizedBox(height: 8),
            _FilterChips(),
            const SizedBox(height: 12),
            _SellerCountIndicator(),
            const SizedBox(height: 12),
            Expanded(child: _SellerList()),
          ],
        ),
      ),
    );
  }
}

class _SearchAndAddHeader extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final viewModel = Provider.of<AppViewModel>(context, listen: false);

    return Row(
      children: [
        Expanded(
          child: TextField(
            onChanged: viewModel.setSellerSearchQuery,
            decoration: InputDecoration(
              hintText: "Cari nama, toko, atau email...",
              prefixIcon: const Icon(Icons.search),
              border: OutlineInputBorder(
                borderRadius: BorderRadius.circular(12),
                borderSide: BorderSide(color: Theme.of(context).colorScheme.surfaceContainerHighest),
              ),
              focusedBorder: OutlineInputBorder(
                borderRadius: BorderRadius.circular(12),
                borderSide: BorderSide(color: Theme.of(context).colorScheme.primary),
              ),
              contentPadding: const EdgeInsets.symmetric(vertical: 0), 
            ),
          ),
        ),
        const SizedBox(width: 12),
        FloatingActionButton(
          onPressed: () => _showAddSellerDialog(context),
          backgroundColor: Theme.of(context).colorScheme.primary,
          foregroundColor: Theme.of(context).colorScheme.onPrimary,
          shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
          elevation: 2,
          child: const Icon(Icons.person_add),
        ),
      ],
    );
  }
}

class _FilterChips extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final viewModel = Provider.of<AppViewModel>(context);
    const filters = ["Semua", "Aktif", "Tidak Aktif", "Banned"];

    return SingleChildScrollView(
      scrollDirection: Axis.horizontal,
      child: Row(
        children: filters.map((status) {
          final selected = viewModel.sellerFilterStatus == status;
          return Padding(
            padding: const EdgeInsets.only(right: 8.0),
            child: FilterChip(
              label: Text(status),
              selected: selected,
              onSelected: (isSelected) {
                if (isSelected) {
                  viewModel.setSellerFilterStatus(status);
                }
              },
              selectedColor: Theme.of(context).colorScheme.primary.withAlpha(15),
              labelStyle: TextStyle(
                color: selected ? Theme.of(context).colorScheme.primary : Theme.of(context).colorScheme.onSurfaceVariant,
              ),
            ),
          );
        }).toList(),
      ),
    );
  }
}

class _SellerCountIndicator extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final count = Provider.of<AppViewModel>(context).sellers.length;
    return Align(
      alignment: Alignment.centerLeft,
      child: Text(
        "Ditemukan $count Penjual",
        style: Theme.of(context).textTheme.bodySmall?.copyWith(
          fontWeight: FontWeight.bold,
          color: Theme.of(context).colorScheme.onSurfaceVariant,
        ),
      ),
    );
  }
}

class _SellerList extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final sellers = Provider.of<AppViewModel>(context).sellers;

    if (sellers.isEmpty) {
      return Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Icon(Icons.people_outline, size: 64, color: Theme.of(context).colorScheme.onSurfaceVariant),
            const SizedBox(height: 8),
            Text("Tidak ada penjual yang cocok", style: TextStyle(color: Theme.of(context).colorScheme.onSurfaceVariant)),
          ],
        ),
      );
    }

    return ListView.separated(
      padding: const EdgeInsets.only(bottom: 24),
      itemCount: sellers.length,
      itemBuilder: (context, index) => _SellerRowItem(seller: sellers[index]),
      separatorBuilder: (context, index) => const SizedBox(height: 12),
    );
  }
}



class _SellerRowItem extends StatelessWidget {
  final Seller seller;

  const _SellerRowItem({required this.seller});

  @override
  Widget build(BuildContext context) {
    final viewModel = Provider.of<AppViewModel>(context, listen: false);
    final colorScheme = Theme.of(context).colorScheme;
    final textTheme = Theme.of(context).textTheme;
    final tealColor = const Color(0xFF26A69A);

    final statusColor = seller.isBanned
        ? colorScheme.error
        : seller.status == "Aktif"
            ? tealColor
            : colorScheme.onSurfaceVariant;

    return Card(
      elevation: 1,
      shadowColor: Colors.black.withAlpha(1),
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(16),
        side: BorderSide(
          color: seller.isBanned ? colorScheme.error.withAlpha(3) : colorScheme.surfaceContainerHighest,
        ),
      ),
      color: seller.isBanned ? colorScheme.errorContainer.withAlpha(1) : colorScheme.surface,
      child: Padding(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          children: [
            Row(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                CircleAvatar(
                  radius: 24,
                  backgroundColor: statusColor.withAlpha(15),
                  child: Text(
                    seller.name.isNotEmpty ? seller.name[0].toUpperCase() : '-',
                    style: textTheme.titleMedium?.copyWith(fontWeight: FontWeight.bold, color: statusColor),
                  ),
                ),
                const SizedBox(width: 16),
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(seller.name, style: textTheme.titleSmall?.copyWith(fontWeight: FontWeight.bold)),
                      Row(
                        children: [
                          Icon(Icons.storefront, size: 14, color: colorScheme.primary),
                          const SizedBox(width: 4),
                          Text(seller.storeName, style: textTheme.bodySmall?.copyWith(fontWeight: FontWeight.bold, color: colorScheme.onSurfaceVariant)),
                        ],
                      ),
                    ],
                  ),
                ),
                _buildMenuButton(context, viewModel),
              ],
            ),
            const SizedBox(height: 12),
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: [
                Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    _buildInfoRow(Icons.email, seller.email, context),
                    const SizedBox(height: 4),
                    _buildInfoRow(Icons.phone, seller.contact, context),
                  ],
                ),
                _buildStatusBadge(context, statusColor),
              ],
            ),
            if (seller.isBanned && seller.banReason != null)
              _buildBanReason(context, colorScheme),
          ],
        ),
      ),
    );
  }

  Widget _buildMenuButton(BuildContext context, AppViewModel viewModel) {
    return PopupMenuButton<String>(
      onSelected: (value) {
        if (value == 'ban') {
          _showBanReasonDialog(context, seller);
        } else if (value == 'unban') {
          viewModel.unbanSeller(seller.id);
        } else if (value == 'delete') {
          viewModel.deleteSeller(seller.id);
        }
      },
      itemBuilder: (BuildContext context) => <PopupMenuEntry<String>>[
        if (!seller.isBanned)
          PopupMenuItem<String>(
            value: 'ban',
            child: ListTile(
              leading: Icon(Icons.block, color: Theme.of(context).colorScheme.error),
              title: Text('Ban Seller', style: TextStyle(color: Theme.of(context).colorScheme.error)),
            ),
          )
        else
          const PopupMenuItem<String>(
            value: 'unban',
            child: ListTile(
              leading: Icon(Icons.check_circle, color: Color(0xFF26A69A)),
              title: Text('Unban Seller', style: TextStyle(color: Color(0xFF26A69A))),
            ),
          ),
        PopupMenuItem<String>(
          value: 'delete',
          child: ListTile(
            leading: Icon(Icons.delete, color: Theme.of(context).colorScheme.error),
            title: Text('Hapus Permanen', style: TextStyle(color: Theme.of(context).colorScheme.error)),
          ),
        ),
      ],
    );
  }

  Widget _buildInfoRow(IconData icon, String text, BuildContext context) {
    return Row(
      children: [
        Icon(icon, size: 12, color: Theme.of(context).colorScheme.onSurfaceVariant),
        const SizedBox(width: 6),
        Text(text, style: Theme.of(context).textTheme.bodySmall?.copyWith(color: Theme.of(context).colorScheme.onSurfaceVariant)),
      ],
    );
  }

  Widget _buildStatusBadge(BuildContext context, Color statusColor) {
    return Container(
      padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 6),
      decoration: BoxDecoration(
        color: statusColor.withAlpha(2),
        borderRadius: BorderRadius.circular(8),
      ),
      child: Text(
        seller.isBanned ? "BANNED" : seller.status.toUpperCase(),
        style: Theme.of(context).textTheme.labelSmall?.copyWith(fontWeight: FontWeight.bold, color: statusColor),
      ),
    );
  }

  Widget _buildBanReason(BuildContext context, ColorScheme colorScheme) {
    return Padding(
      padding: const EdgeInsets.only(top: 8.0),
      child: Container(
        padding: const EdgeInsets.all(10),
        decoration: BoxDecoration(
          color: colorScheme.error.withAlpha(5),
          borderRadius: BorderRadius.circular(8),
        ),
        child: Row(
          children: [
            Icon(Icons.warning, size: 14, color: colorScheme.error),
            const SizedBox(width: 6),
            Expanded(
              child: Text(
                "Alasan: ${seller.banReason}",
                style: Theme.of(context).textTheme.bodySmall?.copyWith(fontSize: 11, color: colorScheme.error),
              ),
            ),
          ],
        ),
      ),
    );
  }
}


// Dialogs
void _showAddSellerDialog(BuildContext context) {
  showDialog(context: context, builder: (ctx) => _AddSellerDialog());
}

class _AddSellerDialog extends StatefulWidget {
  @override
  __AddSellerDialogState createState() => __AddSellerDialogState();
}

class __AddSellerDialogState extends State<_AddSellerDialog> {
  final _formKey = GlobalKey<FormState>();
  final _nameController = TextEditingController();
  final _storeController = TextEditingController();
  final _emailController = TextEditingController();
  final _phoneController = TextEditingController();

  @override
  Widget build(BuildContext context) {
    return AlertDialog(
      title: const Text("Daftarkan Seller Baru", style: TextStyle(fontWeight: FontWeight.bold)),
      content: Form(
        key: _formKey,
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            TextFormField(controller: _nameController, decoration: const InputDecoration(labelText: "Nama Lengkap"), validator: (v) => v!.isEmpty ? "Wajib diisi" : null),
            TextFormField(controller: _storeController, decoration: const InputDecoration(labelText: "Nama Toko"), validator: (v) => v!.isEmpty ? "Wajib diisi" : null),
            TextFormField(controller: _emailController, decoration: const InputDecoration(labelText: "Email"), validator: (v) => v!.isEmpty ? "Wajib diisi" : null),
            TextFormField(controller: _phoneController, decoration: const InputDecoration(labelText: "Nomor Telepon")),
          ],
        ),
      ),
      actions: [
        TextButton(onPressed: () => Navigator.of(context).pop(), child: const Text("Batal")),
        ElevatedButton(
          onPressed: () {
            if (_formKey.currentState!.validate()) {
              Provider.of<AppViewModel>(context, listen: false).addNewSeller({
                'name': _nameController.text,
                'storeName': _storeController.text,
                'email': _emailController.text,
                'contact': _phoneController.text,
              });
              Navigator.of(context).pop();
            }
          },
          child: const Text("Daftarkan"),
        ),
      ],
    );
  }
}

void _showBanReasonDialog(BuildContext context, Seller seller) {
  showDialog(context: context, builder: (ctx) => _BanReasonDialog(seller: seller));
}

class _BanReasonDialog extends StatefulWidget {
  final Seller seller;
  const _BanReasonDialog({required this.seller});

  @override
  __BanReasonDialogState createState() => __BanReasonDialogState();
}

class __BanReasonDialogState extends State<_BanReasonDialog> {
  final _reasonController = TextEditingController();

  @override
  Widget build(BuildContext context) {
    return AlertDialog(
      title: const Text("Tangguhkan Seller", style: TextStyle(fontWeight: FontWeight.bold)),
      content: Column(
        mainAxisSize: MainAxisSize.min,
        children: [
          Text("Konfirmasi pemblokiran toko '${widget.seller.storeName}'. Berikan alasan pemblokiran."),
          TextField(controller: _reasonController, decoration: const InputDecoration(labelText: "Alasan Pelanggaran")),
        ],
      ),
      actions: [
        TextButton(onPressed: () => Navigator.of(context).pop(), child: const Text("Batal")),
        ElevatedButton(
          onPressed: () {
            if (_reasonController.text.isNotEmpty) {
              Provider.of<AppViewModel>(context, listen: false).banSeller(widget.seller.id, _reasonController.text);
              Navigator.of(context).pop();
            }
          },
          style: ElevatedButton.styleFrom(backgroundColor: Theme.of(context).colorScheme.error),
          child: const Text("Tangguhkan"),
        ),
      ],
    );
  }
}
